client/blocks/taxonomy-manager/list.jsx

```diff
@@ -153,6 +160,11 @@ export class TaxonomyManagerList extends Component {
 		} );
 		const hasDefaultSetting = taxonomy === 'category';
 
+		// add a bogus query arg to trigger a new request on delete
+		if ( this.state.deletedItem ) {
+			query.deletedItem = this.state.deletedItem;
```

Comment:
I think we could just write something like `const newQuery = { ...query, deletedItem: this.state.deletedItem }` to avoid mutating the prop.